File: samples/java/hiddenIN2/Numbers.java
//SOLUTION
//IN HIDDEN 3\n-3\n0
//IN 10\n100\n-1\n\u0030
import java.util.Scanner;
public class Numbers
{
   public static void main(String[] args)
   {
      //HIDE
      Scanner in = new Scanner(System.in);
      boolean done = false;
      while (!done)
      {
         //SHOW . . .
         System.out.println("Enter a number, 0 to quit");
         //HIDE
         int n = in.nextInt();
         if (n == 0) done = true;
         else
         {
             int square =  n * n;
             //SHOW . . .
             System.out.println("The square is " + square);
             //HIDE
         }
      }
   }
   //SHOW
}

File: samples/rust/example5/numbers.rs
//SOLUTION
//IN 3\n-3\n0
//IN 10\n100\n-1\n\u0030
use std::io;
fn main() {
    let mut done = false;
    while !done {
        println!("Enter a number, 0 to quit");

        let mut n = String::new();
        io::stdin().read_line(&mut n).expect("failed to readline");
        let mut trimmed = n.trim();
        let mut num = trimmed.parse::<i32>().unwrap();

        if num == 0 {
            done = true;
        } else {
            let mut square = num * num;
            println!("The square is {}", square);
        }
    }
}